File: genetic-api/src/main/java/genetic/api/individual/Evaluated.java
package genetic.api.individual;

import genetic.api.elitism.EliteAcknowledged;
import genetic.api.fitness.Evaluator;
import genetic.api.individual.impl.RatedIndividual;

/**
 * @author piotr.larysz
 */
public interface Evaluated<T1 extends Comparable<T1>, T2> {

    <M extends Evaluator<T1, T2> & EliteAcknowledged> RatedIndividual<T1, T2> evaluate(M evaluator);
}

File: genetic-api/src/main/java/genetic/api/individual/Evaluated.kt
package genetic.api.individual

import genetic.api.elitism.EliteAcknowledged
import genetic.api.fitness.Evaluator
import genetic.api.individual.impl.RatedIndividual

interface Evaluated<T> {

  fun <M> evaluate(evaluator: M): RatedIndividual<T> where M : Evaluator, M : EliteAcknowledged
}